File: leetcode/subArraySumEqualsK.java
class Solution {
  public int subarraySum(int[] nums, int k) {
      
      HashMap<Integer, Integer> map = new HashMap<>();
      
      int ans = 0;
      int sum = 0;
      map.put(0, 1);
      
      for(int num : nums) {
          sum += num;
          int find = sum - k;
          
          if(map.containsKey(find)) ans += map.get(find);
          
          if(map.containsKey(sum)) map.replace(sum, map.get(sum) + 1);
          else map.put(sum, 1);
      }
      
      return ans;
  }
}

File: leetcode/subArraySumEqualsK.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        
        myMap = {0:1}
        partialSum = 0
        ans = 0
        
        for num in nums:
            partialSum += num
            key = partialSum - k
            
            if key in myMap: ans += myMap[key]
            
            if partialSum in myMap: myMap[partialSum] += 1
            else: myMap[partialSum] = 1
            
        return ans